src/Feldspar/Core/Interpretation.hs

```diff
@@ -40,6 +40,33 @@
 
 -- | Defines different interpretations of Feldspar programs
 
+{-# LANGUAGE CPP #-}
+
+#if 0
+#define INCREMENTAL_CSE
+#endif
+
+#ifdef INCREMENTAL_CSE
+module Feldspar.Core.Interpretation where
+
+-- | Possible compilation targets in a broad sense.
```

Comment:
Cumulative and some other things shouldn't conflict with the incremental CSE, put the ifdefs in the export list and around the existing definitions instead.